verifier/src/report_body.rs

```diff
@@ -0,0 +1,177 @@
+        cpu_svn: sgx_cpu_svn_t {
+            svn: [1u8, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16],
+        },
+        misc_select: 17,
+        reserved1: [0u8; 12],
+        isv_ext_prod_id: [1u8, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16],
+        attributes: sgx_attributes_t {
+            flags: 0x0102_0304_0506_0708,
+            xfrm: 0x0807_0605_0403_0201,
+        },
+        mr_enclave: sgx_measurement_t {
+            m: [
+                17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37,
+                38, 39, 40, 41, 42, 43, 43, 44, 45, 46, 47,
+            ],
+        },
+        reserved2: [0u8; 32],
+        mr_signer: sgx_measurement_t {
+            m: [
+                48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68,
+                69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79,
+            ],
+        },
+        reserved3: [0u8; 32],
+        config_id: [
+            80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100,
+            101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
+            118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134,
+            135, 136, 137, 138, 139, 140, 141, 142, 143,
+        ],
+        isv_prod_id: 144,
+        isv_svn: 145,
+        config_svn: 146,
+        reserved4: [0u8; 42],
+        isv_family_id: [
+            147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162,
+        ],
+        report_data: sgx_report_data_t {
+            d: [
+                163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178,
+                179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194,
+                195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210,
+                211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226,
+            ],
+        },
+    };
+
+    #[test]
+    fn report_body_succeeds() {
+        let report_body = ReportBody::from(&REPORT_BODY_SRC);
+        let verifier = ReportBodyVerifier::new(REPORT_BODY_SRC.into(), &report_body);
+        assert_eq!(verifier.verify(), Ok(()));
+    }
+
+    #[test]
+    fn report_body_fail() {
+        let report_body = ReportBody::from(&REPORT_BODY_SRC);
+        let mut report_body_src = REPORT_BODY_SRC;
+        report_body_src.attributes.flags = 0;
+        let report_body_src: ReportBody = report_body_src.into();
+        let verifier = ReportBodyVerifier::new(report_body_src.clone(), &report_body);
+        assert_eq!(
+            verifier.verify(),
+            Err(VerificationError(format!(
+                "Attributes mismatch: expected {:?}, got {:?}",
+                report_body_src.attributes(),
+                report_body.attributes()
+            )))
```

Comment:
I think this exposes a current weakness in my desire to have a generic `AttributesVerifier` that doesn't care which report it's part of. This error message still causes a client to dig around and find where the incorrect attributes are at.
I'm thinking either I need to get on board with the `QeReportAttributesVerifier` and `AppReportAttributesVerifier` or the `ReportBody` needs to capture child errors and further decorate. `"Quoting enclave, ReportBody, Attributes mismatch: expected ...."`

Reply:
Captured improvements in #8